WIP - Fix Desktop app Mac target notarization

mokagio opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to internal ref pdnsEh-1OD-p2

## Proposed Changes

* Updates the `after_sign_hook.js` script to successfully notarize using the new `@electron/notarize` API.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

mokagio committed:
desktop: Add required `teamId` parameter to notarization call

mokagio commented on desktop/bin/after_sign_hook.js:
```diff
@@ -38,6 +38,7 @@ module.exports = async function ( context ) {
 		appleId: NOTARIZATION_ID,
 		appleIdPassword: NOTARIZATION_PWD,
 		ascProvider: NOTARIZATION_ASC_PROVIDER,
+		teamId: NOTARIZATION_ASC_PROVIDER,
```
With this change, notarization succeeds on my machine

<img width="1279" alt="image" src="https://github.com/user-attachments/assets/ec6a3560-637a-4fc1-a67d-feb7189e6960">

However, it might not be enough to work in CI because of a possible lack of code signing identity there (required upstream of notarization)

@wojtekn what is the best way for me to test this? Is it okay to spam the beta channel with new betas, or should I hack the script and CI to attempt the Mac build on every commit without publishing? My preference would be for the latter, to avoid release noise.

wojtekn replied:
@mokagio It's okay to spam betas until we get this fixed. The build process won't publish a new beta in https://github.com/Automattic/wp-desktop/releases if it doesn't reach `wp-desktop-publish` step.

> However, it might not be enough to work in CI because of a possible lack of code signing identity there (required upstream of notarization)

As far as I understand the error message, on CI notarize fails because the code is not signed.

mokagio committed:
Disable branch filters to test macOS build

mokagio committed:
Disable check to conditionally build artifacts

mokagio committed:
Always notarize

wp-desktop reviewed (dismissed):
WordPress Desktop CI Failure for job "wp-desktop-mac".

@mokagio please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/1161310). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

mokagio committed:
Force code signing

mokagio committed:
Set up Ruby tooling

mokagio committed:
Run bundle install in CircleCI

mokagio committed:
Run bundle install in desktop/ + User rbenv

mokagio committed:
Install Ruby 3.2.2 in CircleCI

mokagio committed:
Downgrade to Ruby 3.1.3 because is available in CircleCI image

mokagio committed:
Attempt to build with latest Xcode, 15.4.0

mokagio committed:
Disable code signing to see if build works

mokagio committed:
Set up Fastlane to manage code signing certs

mokagio committed:
Run lane to configure code signing in CI

mokagio committed:
Revert "Disable code signing to see if build works"

This reverts commit 06aa7bab93c759bfb706258f6c4f84b709dc2364.

mokagio committed:
Add gem to attempt fixing CircleCI-specific CI issue

See https://github.com/fastlane/fastlane/issues/21794

mokagio committed:
Skip decrypting p12

mokagio committed:
Remove CSC_LINK, see if it picks up the right certificate

mokagio committed:
Actually skip decrypting Mac p12 in CI...

Previous attempt was for local build only?

mokagio committed:
Switch to different org to see if it addresses CSSMERR_TP_NOT_TRUSTED

See failure at
https://app.circleci.com/pipelines/github/Automattic/wp-calypso/240254/workflows/dfe62864-ee4e-4c27-a7f9-02077e64f9ba/jobs/1161452?invite=true#step-107-14247_88

mokagio commented:
Closing in favor of https://github.com/Automattic/wp-calypso/pull/93845
